### PR TITLE
Add section 4 translation

### PR DESCRIPTION
**Pull request by 21zhouyun**
Finished everything up to 'Striking a balance between informative and affordable experiments'.

### CODE REVIEW

**Comment by 21zhouyun** [comment]
@RangeKing Please take a look.

**Comment by RangeKing** [comment]
> @RangeKing Please take a look.

Thank you for your contrbution! I'll review ASAP.

File: README_zh-CN.md [hunk 1]
```diff
@@ -225,110 +225,110 @@
...
+-   有足够的计算资源来进行调参实验，至少能够并行发起数个训练流程。
 
 ### 增量调整策略
 
-***概括：*** *从简单的配置开始，逐步进行改进，同时深入了解问题。确保任何改进都基于强有力的证据，以避免增加不必要的复杂性。*
+***概括：*** *从简单的配置开始，循序渐进，同时进一步了解问题。确保任何改进都有据可循，以避免增加不必要的复杂度。*
 
--   我们的最终目标是找到一种配置来最大化我们模型的性能。
+-   我们的最终目标是找到一种训练配置来最大化我们模型的性能。
     -   在某些情况下，我们的目标是在固定截止日期（例如提交给竞赛）之前最大限度地改进模型。
```
Comment (RangeKing) [inline README_zh-CN.md]:
```suggestion
    -   在某些情况下，我们的目标是在固定截止日期（例如竞赛提交期限）之前最大限度地改进模型。
```

File: README_zh-CN.md [hunk 2]
```diff
@@ -225,110 +225,110 @@
...
--   原则上，我们可以通过使用算法自动搜索可能配置的整个空间来最大化性能，但这不是一个实用的选择。
-    -   可能配置的空间非常大，目前还没有任何算法足够复杂，可以在没有人工指导的情况下有效地搜索这个空间。
--   大多数自动搜索算法依赖于定义要搜索的配置集的手工设计的*搜索空间*，这些搜索空间可能非常重要。
--   最大化性能的最有效方法是从简单的配置开始，逐步添加功能并进行改进，同时深入了解问题。
-    -   我们在每一轮调整中都使用自动搜索算法，并随着我们理解的增长不断更新我们的搜索空间。
+-   原则上，我们可以使用算法自动搜索整个配置空间来最大化性能，但实践中这往往不实际。
+    -   配置空间可能非常大，目前还没有任何算法可以在没有人工指导的情况下有效地搜索这个空间。
+-   大多数自动搜索算法依赖于人工设计的*搜索空间*，这些搜索空间往往非常重要。
+-   更有效的方法是从简单的配置开始，逐步添加功能并进行改进，同时深化对问题的理解。
+    -   我们在每一轮调整中都使用自动搜索算法，并随着我们理解的深度不断更新我们的搜索空间。
```
Comment (RangeKing) [inline README_zh-CN.md]:
```suggestion
    -   我们在每一轮调整中都使用自动搜索算法，并随着我们理解的深入不断更新我们的搜索空间。
```

File: README_zh-CN.md [hunk 3]
```diff
@@ -363,45 +363,45 @@
...
+    1.  比较足够多目标超参数的不同值。
+    2.  在足够大的搜索空间上调整冗余超参数。
+    3.  对冗余超参数的搜索空间进行足够密集的采样。
+-   这三个必要条件，确保我们能从实验中获得足够多的经验。
+    -   尽可能多地比较目标超参数的值可以拓宽我们从实验中获得的经验的范围。
+    -   包括尽可能多的有冗余超参数，并允许每个冗余超参数有尽可能大的值域。这样我们更有信心相信，对于一个目标超参数配置，在当前搜索空间中，**存在**“好的”冗余参数。
+        -   否则，我们可能因为没有搜索某些冗余超参数空间，而对不同目标超参数值进行了不公平的比较。
+    -   尽可能密集地对冗余超参数采样，也能让我们更加有信心相信搜索流程能够找到好的冗余超参数配置。
+        -   否则，我们可能对不同目标超参数值进行了不公平的比较。因为某些值恰好有更好的冗余超参数配置。
+-   不幸的是，对这三个要求的*任何*一个进行改进都需要增加试验次数，从而增加资源成本。或者，我们只能从其他要求中想办法节约资源。
```
Comment (RangeKing) [inline README_zh-CN.md]:
```suggestion
-   不幸的是，对这三个要求的*任何*一个进行改进都需要增加试验次数，从而增加资源成本。或者，想办法降低其他要求，以节约资源。
```

File: README_zh-CN.md [hunk 4]
```diff
@@ -500,63 +500,63 @@
...
 <br>
 
--   生成图的努力越多，我们就越不可能尽可能多地查看它们，因此我们有必要设置我们的基础设施以自动生成尽可能多的图。
--   至少，我们会为我们在实验中变化的所有超参数自动生成基本超参数轴图。
--   此外，我们会自动为所有试验生成训练曲线，并尽可能轻松地找到每项研究中最好的少数试验并检查它们的训练曲线。
--   我们可以添加许多其他有用的潜在图表和可视化。尽管上面描述的是一个很好的起点，但套用杰弗里 · 辛顿 (Geoffrey Hinton) 的话，“每次设计新事物时，您都会学到新事物。”
+-   绘图越困难，我们看得就越少。所以我们应该尽可能将绘图自动化。
+-   我们至少应该为试验中用到的超参数自动生成基本超参数轴图。
+-   此外，我们会自动为所有试验生成训练曲线，并确保我们能够很轻松的找到每次研究的最佳试验和对应试验的训练曲线。
+-   除此之外，还有许多其他有用的图表和可视化方法。上面描述的图表是很好的起点，但套用杰弗里 · 辛顿 (Geoffrey Hinton) 的话，“每当你绘制新事物时，你都会学到一些新知识。”
```
Comment (RangeKing) [inline README_zh-CN.md]:
```suggestion
-   上面描述的图表是很好的起点。但除此之外，还有许多其他有用的图表和可视化方法。就像杰弗里 · 辛顿 (Geoffrey Hinton) 说的，“每当你绘制新图表时，都能学到一些新知识。”
```

File: README_zh-CN.md [hunk 5]
```diff
@@ -225,110 +225,110 @@
...
+-   在设计新一轮实验时，我们根据实验目的确定目标超参数。
+    -   在此阶段，我们将所有其他超参数视为冗余超参数。
+-   接下来，我们将一些冗余超参数转作为固定超参数。
+    -   如果有无限的计算资源，我们会将所有非目标超参数保留为冗余超参数，这样我们从实验中得出的结论就不会受到固定超参数的限定。
+    -   然而，冗余超参数越多，我们没能充分针对每个目标超参数调优冗余超参数的风险就越高，从而我们从实验中得出错误结论的风险也越高。
+        -   如[下文](#平衡实验的信息量和成本)所述，我们可以通过增加计算资源来应对这种风险，但通常我们的最大资源预算低于调整所有非目标超参数所需的计算资源。
+    -   当我们判断将一个冗余超参数转换为固定超参数所带来的限制少于调优它所需的计算资源时，我们可以进行这种转换。
+        -   一个冗余超参数和目标超参数的相互影响越多，固定这个参数所带来的限制就越多。例如，权重衰减强度的最佳值通常取决于模型大小，因此固定权重衰减的强度来比较不同的模型大小，往往得不出有效的结论。
+-   尽管超参数的类型取决于实验目标，但对于某些类别的超参数，我们有以下经验法则：
+    -   在各种优化器超参数（例如学习率、动量、学习率调度参数、Adam优化器的beta等）中，至少有一些是冗余超参数，因为它们往往与其他变化相互影响。
```
Comment (RangeKing) [inline README_zh-CN.md]:
```suggestion
    -   在各种优化器超参数（例如学习率、动量、学习率调度参数、Adam 优化器的 beta 等）中，至少有一些是冗余超参数，因为它们往往与其他变化相互影响。
```

File: README_zh-CN.md [hunk 6]
```diff
@@ -500,63 +500,63 @@
...
--   我们的探索工作应该已经揭示了最重要的要调整的超参数（以及它们的合理范围），我们可以使用这些超参数来构建搜索空间，以使用尽可能大的调整预算进行最终的自动调整研究。
--   由于我们不再关心最大化我们对调整问题的洞察力，许多[准随机搜索的优点](#why-use-quasi-random-search-instead-of-more-sophisticated-black-box-optimization-algorithms-during-the-exploration-phase-of-tuning)不再适用，应该使用贝叶斯优化工具来自动找到最佳超参数配置。
+-   我们的优先级总有一天会从更多了解参数调优问题转变到找到一个最佳的配置上。。
+-   这时，我们应该已经有了一个包含着已知最佳配置的搜索空间。
+-   我们的探索工作应该已经揭示了最重要的超参数（以及它们的合理范围），我们可以使用这些超参数来构建搜索空间，并使用尽可能大的计算预算进行最终的自动搜参研究。
+-   由于我们已经不需要更进一步了解调优问题本身了，许多[准随机搜索的优点](#为什么在优化的探索阶段使用准随机搜索而不是更复杂的黑盒优化算法？)不再适用，应该使用贝叶斯优化工具来自动找到最佳超参数配置。
     -   如果搜索空间包含大量发散点（获得 NaN 训练损失或什至训练损失比均值差很多标准差的点），使用黑盒优化工具来正确处理发散试验很重要（请参阅[具有未知约束的贝叶斯优化](https://arxiv.org/abs/1403.5607)是处理此问题的绝佳方法）。
 -   此时，我们还应该考虑检查测试集上的性能。
-    -   原则上，我们甚至可以将验证集折叠到训练集中，并重新训练通过贝叶斯优化找到的最佳配置。但是，这仅适用于未来不会针对此特定工作负载发布的情况（例如，一次性 Kaggle 竞赛）。
+    -   原则上，我们甚至可以将验证集放到训练集中，并重新训练通过贝叶斯优化找到的最佳配置。但是，这仅适用于未来不会再次上线的情况（例如，一次性 Kaggle 竞赛）。
```
Comment (RangeKing) [inline README_zh-CN.md]:
```suggestion
    -   原则上，我们甚至可以将验证集放到训练集中，并重新训练通过贝叶斯优化找到的最佳配置。但是，这仅适用于未来不会再次上线的情况（例如，Kaggle 竞赛）。
```

**Review by RangeKing — COMMENTED** [review]
是不是可以把`实验`和`试验`两个词语统一？

**Comment by 21zhouyun** [comment]
> 是不是可以把`实验`和`试验`两个词语统一？

done

**Review by RangeKing — APPROVED** [review]
LGTM

**Pull request merged**